src/gucs/embedding.rs

```diff
@@ -0,0 +1,54 @@
+use std::ffi::CStr;
+
+use pgrx::{GucContext, GucFlags, GucRegistry, GucSetting};
+
+use crate::prelude::guc_parse_failed;
+
+fn guc_string_parse(target: &'static GucSetting<Option<&'static CStr>>, name: String) -> String {
```

Comment:
We can take advantage of a checker functions for every string GUC to ensure it's valid UTF-8.